### PR TITLE
get rid of the redundancy field in kube-proxy option

### PR DESCRIPTION
**Pull request by stewart-yu**
**What type of PR is this?**
 /kind cleanup

**What this PR does / why we need it**:
Two filed exists in kube-proxy Option, but we can reference all in config scheme file
```
scheme *runtime.Scheme
codecs serializer.CodecFactory
```
remove those

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kube-proxy/app/server.go [hunk 1]
```diff
@@ -54,7 +53,7 @@ import (
 	"k8s.io/kubernetes/pkg/proxy"
 	"k8s.io/kubernetes/pkg/proxy/apis"
 	kubeproxyconfig "k8s.io/kubernetes/pkg/proxy/apis/config"
-	"k8s.io/kubernetes/pkg/proxy/apis/config/scheme"
+	kubeproxyconfigscheme "k8s.io/kubernetes/pkg/proxy/apis/config/scheme"
```
Comment (JieJhih) [inline cmd/kube-proxy/app/server.go]: Wonder if it should use ``kubeproxyscheme`` instead? It is shorter.

Comment (stewart-yu) [reply cmd/kube-proxy/app/server.go]: Indeed. but `kubeproxyconfigscheme` sync with using in `controller-manager`, and indicate the `scheme` belong to `proxy Componentconfig`. i afraid of that `kubeproxyscheme` missing the config message

**Comment by vllry** [comment]
Do we want to be using globals (from `scheme`?) in this way?

**Comment by stewart-yu** [comment]
> Do we want to be using globals (from `scheme`?) in this way?

may be you misunderstand it, not `globals`. 
 In `Option` struct, two field
```
scheme *runtime.Scheme
codecs serializer.CodecFactory
```
seems redundancy, no need.
we just reference the `scheme ` and `codecs ` with the `config`  once discard those field, the value is not change. please the the PR change

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**